api/management/commands/create_groups.py

```diff
@@ -0,0 +1,34 @@
+from collections.abc import Iterable
+from typing import Any
+
+from django.core.management.base import BaseCommand, CommandError, CommandParser
+
+from api.models import Teacher
+from api.processors.auxil.algorithms import GroupBuilderAlgorithm
+
+
+class Command(BaseCommand):
+    """
+    Triggers automatic group creation.
+    If teacher_ids are not specified, triggers for all available teachers.
+    """
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument("--teacher_ids", nargs="*", type=int)
+
+    def handle(self, *_: str, **options: Any) -> None:
+        teachers: Iterable[Teacher] = []
+        if options["teacher_ids"]:
+            teachers = [
+                Teacher.objects.get(personal_info__id=teacher_id)
+                for teacher_id in options["teacher_ids"]
+            ]
+            for teacher in teachers:
+                if not GroupBuilderAlgorithm.is_teacher_available(teacher):
+                    raise CommandError(f"Teacher is not available: {teacher}")
+        else:
+            # Get all available teachers if none specified
+            teachers = GroupBuilderAlgorithm.get_available_teachers()
+
+        for teacher in teachers:
+            GroupBuilderAlgorithm.create_and_save_group(teacher.personal_info.pk)
```

Comment:
I think somewhere here should be a POST request to my bot? If it seems like the correct spot, we could add a TODO here that this is where a webhook call will be made.

Reply:
If you can provide references for later, this will be great (which particular request?)

Reply:
Well I don't know the route yet, but it will be a request that signals the bot that it must send notifications to students and teacher and collect their reactions on whether they are ready to study/teach.